optima/model.py

```diff
@@ -379,10 +374,16 @@ def model(simpars=None, settings=None, verbose=None, benchmark=False, die=True):
             pop2 = this['pop2']
             thistrans = this['trans']
             
-            thisforceinf = 1 - mpow((1-thistrans*circeff[pop1,t]*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
-            forceinfvec[pop1] = 1 - (1-forceinfvec[pop1]) * (1-thisforceinf)  
-            
-            if not(forceinfvec[pop1]>=0):
+            if male[pop1]: # Separate FOI calcs for circs vs uncircs
+                thisforceinf_uncirc = 1 - mpow((1-thistrans*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
+                thisforceinf_circ = 1 - mpow((1-thistrans*circeff[pop1,t]*prepeff[pop1,t]*stieff[pop1,t]), (dt*cond*acts*effhivprev[pop2]))
```

Comment:
this could be done with a loop but it's a bit complicated to get the conditionals right so will leave for now